code/game/machinery/bots/medbot.dm

```diff
@@ -507,7 +531,23 @@
 	visible_message("<span class='userdanger'>[src] blows apart!</span>")
 	var/turf/Tsec = get_turf(src)
 
-	new /obj/item/weapon/storage/firstaid(Tsec)
+	switch(skin)

```

Comment:
Yeah, using a switch statement on a text that contains the variant of medibot is horrible, although it doesn't look like it can be done any other way without refactoring how medibot sprites are handled entirely.